In [27]:
import pandas as pd
import numpy as np
import scipy.stats as st
import math

pd.set_option('display.max_columns', 30)
df=pd.read_csv('S1_Dataset.csv')
df

id     sex  age (months)  dlmo time  \
0                              1  female          33.7      19.24   
1                              2  female          31.5      18.27   
2                              3    male          31.9      19.14   
3                              4  female          31.6      19.69   
4                              5  female          33.0      19.52   
5                              6  female          36.2      18.22   
6                              7    male          36.3      19.28   
7                              8    male          30.0      21.06   
8                              9    male          33.2      19.38   
9                             10  female          37.1      19.93   
10                            11    male          32.9      18.79   
11                            12  female          35.0      19.65   
12                            13    male          35.1      19.83   
13                            14  female          35.6      19.88   
14                            15  female          36.6      19.94   
15                            16    male          36.5      20.25   
16                            17  female          33.7      20.33   
17                            18    male          36.4      20.16   
18                            19  female          33.6      19.68   
19                            20    male          33.8      20.51   
20                           NaN     NaN           NaN        NaN   
21               0 = no; 1 = yes     NaN           NaN        NaN   
22  all times are decimal values     NaN           NaN        NaN   

    days napped  napping  nap lights outl time  nap sleep onset  nap midsleep  \
0           0.0      0.0                   NaN              NaN           NaN   
1           0.0      0.0                   NaN              NaN           NaN   
2           0.0      0.0                   NaN              NaN           NaN   
3           0.0      0.0                   NaN              NaN           NaN   
4           0.0      0.0                   NaN              NaN           NaN   
5           4.0      1.0                 14.00            14.22         15.00   
6           1.0      1.0                 14.75            15.03         15.92   
7           5.0      1.0                 13.09            13.43         14.44   
8           2.0      1.0                 14.41            14.42         15.71   
9           3.0      1.0                 13.12            13.42         14.31   
10          4.0      1.0                 13.99            14.03         14.85   
11          5.0      1.0                 13.18            13.45         14.33   
12          3.0      1.0                 13.94            14.48         15.26   
13          4.0      1.0                 12.68            13.08         13.92   
14          4.0      1.0                 12.71            12.88         13.80   
15          3.0      1.0                 13.74            14.68         15.66   
16          5.0      1.0                 13.15            13.87         14.49   
17          5.0      1.0                 12.47            12.56         13.30   
18          3.0      1.0                 14.71            14.85         15.46   
19          3.0      1.0                 12.68            13.54         14.30   
20          NaN      NaN                   NaN              NaN           NaN   
21          NaN      NaN                   NaN              NaN           NaN   
22          NaN      NaN                   NaN              NaN           NaN   

    nap sleep offset  nap wake time  nap duration  nap time in bed  \
0                NaN            NaN           NaN              NaN   
1                NaN            NaN           NaN              NaN   
2                NaN            NaN           NaN              NaN   
3                NaN            NaN           NaN              NaN   
4                NaN            NaN           NaN              NaN   
5       

In [24]:
#Our goal is to analyze the bedtime of napping and non-napping toddlers.
#night_bedtime=df['night bedtime']
#daytime_nap=df.loc[df.napping==1]
bedtime_nap=df['night bedtime'].loc[df['napping']==1]
bedtime_no_nap=df['night bedtime'].loc[df['napping']==0]
print(len(bedtime_nap))
print(len(bedtime_no_nap))





15
5


# Formula for the interval estimation(finding out the CONFIDENCE INTERVAL)
Interval_Estimate= sample_mean+/-MOE
MOE=z alpha/2 * sigma/root(n)     
#if n>30, z distribution --> norm.interval() function is used

#if n<30, t ditribution ---> t.interval() function is used

need to find dof(degree of freedom)

#The arguments for alpha=CI percentage, df = degree of freedom, scale = SEM(standard error of mean), loc = mean

In [ ]:
#nap_mean_bedtime = bedtime_nap.mean() --> sample mean
#nap_s_bedtime = np.std(bedtime_nap,ddof=1) -->standard deviation
#nap_se_mean_bedtime = nap_s_bedtime/math.sqrt(len(bedtime_nap)) ---> standard error
#nap_t_star = t.ppf(0.975,df=14) --->to find talpha/2 for z value with 95% CI=>0.975 and degree of freedom=14

#values for alpha=0.05,0.025
#z_alpha_NormalDistribution={0.025:1.96,0.05:1.68}
#z_alpha_TDistribution={0.025:2.131,0.05:1.75}

In [40]:


CIbedtimeForNaptaken=st.t.interval(alpha=0.95, df=len(bedtime_nap)-1, loc=np.mean(bedtime_nap), scale=st.sem(bedtime_nap)) 

CIbedtimeForNapNottaken=st.t.interval(alpha=0.95, df=len(bedtime_no_nap)-1, loc=np.mean(bedtime_no_nap), scale=st.sem(bedtime_no_nap)) 
#st.norm.interval(alpha=0.95, loc=np.mean(bedtime_nap), scale=st.sem(bedtime_nap))
print(CIbedtimeForNaptaken)
print(CIbedtimeForNapNottaken)



    

(19.976680775477412, 20.631319224522585)
(18.95974084563192, 20.220259154368087)


# Interpretation: 

From the above results, we conclude that we are 95% confident that the average bedtime for napping toddlers is between the time 19.98 – 20.63 (pm) while for non-napping toddlers it is between 18.96 – 20.22 (pm). These results are as per our expectation that if you take a nap during the day you will sleep late at night.

In [56]:


def interval_estimate(n,data):
    try:
        if (n>=30):
            CI=st.norm.interval(alpha=0.95, loc=np.mean(data), scale=st.sem(data))
        else:
            CI=st.t.interval(alpha=0.95, df=len(data)-1, loc=np.mean(data), scale=st.sem(data))
            #print(CI)
        return CI
    except:
        print("check the type of estimate an dits proper calculation")
            


In [59]:
n=50
data = np.random.randint(10, 30, n) # in the range of 10 to 30 take 50 count
CI=interval_estimate(n,data)  
print(f"The confidence interval is {CI}")

The confidence interval is (18.912927644614907, 21.807072355385092)
